plugins/modules/mso_schema_site_contract_service_graph.py

```diff
@@ -0,0 +1,354 @@
+from ansible_collections.cisco.mso.plugins.module_utils.schema import MSOSchema
+from ansible_collections.cisco.mso.plugins.module_utils.mso import MSOModule, mso_argument_spec, mso_schema_site_contract_service_graph_spec
+
+
+def main():
+    argument_spec = mso_argument_spec()
+    argument_spec.update(
+        tenant=dict(type="str"),
+        schema=dict(type="str", required=True),
+        template=dict(type="str", required=True),
+        contract=dict(type="str"),
+        site=dict(type="str", required=True),
+        service_graph_schema=dict(type="str"),
+        service_graph_template=dict(type="str"),
+        service_graph=dict(type="str"),
+        node_relationship=dict(type="list", elements="dict", options=mso_schema_site_contract_service_graph_spec()),
+        state=dict(type="str", default="present", choices=["absent", "present", "query"]),
+    )
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+        required_if=[
+            ["state", "absent", ["contract"]],
+            ["state", "present", ["contract", "service_graph"]],
+        ],
+    )
+
+    tenant = module.params.get("tenant")
+    schema = module.params.get("schema")
+    template = module.params.get("template").replace(" ", "")
+    contract = module.params.get("contract")
+    site = module.params.get("site")
+    service_graph_schema = module.params.get("service_graph_schema")
+    service_graph_template = module.params.get("service_graph_template")
+    service_graph = module.params.get("service_graph")
+    node_relationship = module.params.get("node_relationship")
+    state = module.params.get("state")
+
+    mso = MSOModule(module)
+
+    mso_schema = MSOSchema(mso, schema, template)
+    mso_schema.set_template(template)
+
+    service_graph_schema_id = schema_id = mso.lookup_schema(schema, True)
+    service_graph_reference_schema = service_graph_schema or schema
+
+    # Get service graph reference schema id if the service graph schema is not matching with the schema
+    if service_graph_reference_schema != schema:
+        service_graph_schema_id = mso.lookup_schema(service_graph_reference_schema, True)
+        if service_graph_schema_id is None:
+            mso.fail_json(msg="Provided service_graph_schema: '{0}' does not exist.".format(service_graph_schema))
+
+    service_graph_reference_schema_id = service_graph_schema_id or schema_id
+    service_graph_reference_template = service_graph_template or template
+
+    # Get site id
+    site_id = mso.lookup_site(site)
+    mso_schema.set_site(template, site)
+    mso_schema.set_site_contract(contract, False)
+
+    if mso.platform == "on-premise" and (node_relationship is None or tenant is None) and state == "present":
+        # When site type is on-premise, both node_relationship and tenant are required
+        mso.fail_json(msg="Missing either node_relationship or tenant required attribute.")
```

Comment:
Do you want to provide in output what the user has provided, and only display those that are None?